birdhouse/README.rst

```diff
@@ -114,6 +114,37 @@ Disk: at least 100 TB, depending how much data is hosted on Thredds and Geoserve
 In general, the more users, the more cpu cores and memory needed.  The more data, more memory and bigger and faster disks needed.
 
 
+Local deployment for development
+--------------------------------
+
+To deploy the birdhouse stack for development, a self-signed SSL certificate is required:
+
+First create a self signed certificate as described in the `LetsEncrypt documentation <https://letsencrypt.org/docs/certificates-for-localhost/#making-and-trusting-your-own-certificates>`_ .
+Then add the certificate you just generated to the trust store as described `here <https://ubuntu.com/server/docs/security-trust-store>`_ .
+Finally, combine the generated key and certificate files together to use as the `SSL_CERTIFICATE` for this stack.
+
+The following should work to generate a self-signed ssl certificate on ubuntu 20.04+:
+
+.. code-block::
+
+    openssl req -x509 -out localhost.crt -keyout localhost.key \
+      -newkey rsa:2048 -nodes -sha256 \
+      -subj '/CN=localhost' -extensions EXT -config <( \
+       printf "[dn]\nCN=localhost\n[req]\ndistinguished_name = dn\n[EXT]\nsubjectAltName=DNS:localhost\nkeyUsage=digitalSignature\nextendedKeyUsage=serverAuth")
+
+    # https://ubuntu.com/server/docs/security-trust-store
+    sudo cp localhost.crt /usr/local/share/ca-certificates/
+
+    cat localhost.crt localhost.key > cert.pem
```

Comment:
I think the generated `cert.pem` file needs to be `chmod 0600`, otherwise `nginx` complains in some situations about invalid permissions.

Reply:
the nginx docker container runs all workers as root so there is no permission limits enforced

Reply:
I don't think it is a problem in the sense of accessing/reading the file. It is nginx or the underlying SSL libraries themselves that flag the permissions as invalid, as it is prone to security problems. Running as root is valid, but it should be only root that can read it (no groups/others), and for security, it should not be modifiable from within the container, so `600` + mount with `:ro`.